caravel/assets/javascripts/modules/caravel.js

```diff
@@ -76,16 +77,13 @@ const px = function () {
       container,
       containerId,
       selector,
-      querystring(params) {
-        const newParams = params || {};
+      querystring() {
         const parser = document.createElement('a');
         parser.href = data.json_endpoint;
-        if (dashboard !== undefined) {
-          let flts = '';
-          if (newParams.extraFilters !== false) {
-            flts = dashboard.effectiveExtraFilters(sliceId);
-            flts = encodeURIComponent(JSON.stringify(flts));
-          }
+        if (controller.type === 'dashboard') {
```

Comment:
1. origJsonEndpoint with the params for the chart is only used in the dashboard to be able to remove the extra_filters. Hence, I would save it in the dashboard.
2. The form input with the params for the chart is only used in explore.
3. In the default case, e.g. standalone, the query params are from the slide itself. 
Thus, I would put the query string generation in the controller interface in use the default case when it is not implemented.